skills_ml/algorithms/occupation_classifiers/classifiers.py

```diff
@@ -8,29 +14,56 @@
 import pickle
 import re
 
-def convert(name):
-    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
-    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+first_cap_re = re.compile('(.)([A-Z][a-z]+)')
+all_cap_re = re.compile('([a-z0-9])([A-Z])')
+
+def convert_camel_to_lower(name):
+    s1 = first_cap_re.sub(r'\1_\2', name)
+    return all_cap_re.sub( r'\1_\2', s1).lower()
+
 
 class SocClassifier(object):
-    """ Interface of SOC Code Classifier.
+    """ Interface of SOC Code Classifier for computer class to use.
     """
 
     def __init__(self, classifier):
         self.classifier = classifier
 
-    def predict_soc(self, tokenized_list):
-        return self.classifier.predict_soc(tokenized_list)
+    def predict_soc(self, tokenized_words):
+        return self.classifier.predict_soc(tokenized_words)
 
     @property
     def name(self):
-        return "soc_" + convert(self.classifier.name)
+        return "soc_" + convert_camel_to_lower(self.classifier.name)
 
     @property
     def description(self):
         return f"SOC code classifier using {self.classifier.description}"
 
 
+class CombinedClassifier(object):
+    def __init__(self, embedding, classifier, **kwargs):
+        self.embedding = embedding
+        self.classifier = classifier
+        self.combined = Pipeline([
+            ('tokens_to_vector', EmbeddingTransformer(self.embedding)),
+            ('classify', self.classifier)
+        ])
+        self.soc_encoder = SocEncoder(list(majorgroupname.keys()))
```

Comment:
Is this tying the classifier too closely to the major group? In the training code there are separate code paths for major group and full soc but here it's hardcoded.